openedx/core/djangoapps/content_staging/models.py

```diff
@@ -144,3 +144,43 @@ def save(self, *args, **kwargs):
         # Enforce checks on save:
         self.full_clean()
         return super().save(*args, **kwargs)
+
+
+class UserLibrarySync(models.Model):
+    """
+    Each user can trigger a sync from a library component to that component in a course.
+    This model is used to facilitate that and to ease tracking.
```

Comment:
Tracking will be implemented by `PublishableEntityLink` in https://github.com/openedx/openedx-learning/pull/269 so this seems to overlap with that?


Actually I'm wondering if we really need `UserLibrarySync` at all. I think you can implement this whole thing just using the existing `StagedContent` model. 


In your `import_staged_content_for_library_sync` code:

```diff
-library_sync = content_staging_api.save_xblock_to_user_library_sync(lib_block, request.user.id)
+staged_content = content_staging_api.stage_xblock_temporarily(lib_block, request.user.id)

try:
    # Sync static assets
finally:
    staged_content.delete()
```

Reply:
I removed user library sync. sorry it was an artifact of a first version where I used a rest api endpoint from the frontend but after simplifying I didn't think of removing it.